### ISSUE TITLE
UTM projection

### ISSUE THREAD

**Issue by mpadge** [hypertidy/geoarea#1]
@mdsumner For this to be a viable package idea, I reckon we need the two current methods plus a third, `reproj`-type method to UTM or equivalent. Two questions:

1. What's your read of current state of `proj`/`reproj`? Dewey's `projlib` attempt just bundled `proj`, but has been kicked off CRAN. It should be relatively straightforward to bundle a single projection in here and use that - thoughts?
2. What do you think would be the most useful projection to use? UTM has obvious disavantages, and the error tolerance, especially when crossing bands, is maybe a little too high to be truly useful for attempting to accurately measure areas. Any other ideas?

**Comment by mdsumner** [comment]
I was actually working on this, there are two interesting cases - a local laea for area, I advise this as a good general case anyway, and aeqd for distance - I've long wanted to vectorize the aeqd params for individual paired distances (along a path or simply many to one).

Start with spherical then worry about ellipsoidal, needs rotation for longlat to local radians, then the code you see in proj/src/projections 😄 - the P-> or Q-> carries the centring params, and Mode N_POLE, EQUIT, etc signify special cases for the central point that simplify the code (because sin/cos form is zero).

Otherwise, I'm using proj4 via {reproj} (reproj_xy is in github only and I'm getting used to having it available), PROJ could be resurrected given my learnings recently.

I think a simplistic projections package, just laea, aeqd, mercator, utm, lcc, geocentric would be worth doing (less accurate than ballpark, idealised earth, etc) and I've been pondering it for a long time 🙏

**Comment by mpadge** [comment]
Yes indeed, a simplified projections package would be awesome here! The spherical stuff is what is already in here in [the "cheap" code](https://github.com/hypertidy/geoarea/blob/45b2efe11ecf198b283f39c9c906ed4eddfb1837/src/geoarea.c#L53), so spherical projections aren't worthwhile given that. But simple laea/utm-type projections only would be extremely useful. Obviously constant ongoing wrangling of `proj` is a nightmare enough to get `libproj` kicked off, so there's a definite use case for a simplified version. Ping me elsewhere if you want me to chip in, and maybe i won't worry too much more here for the moment. Thanks!

**Comment by mdsumner** [comment]
I wrapped up  [GeographicLib](https://geographiclib.sourceforge.io/C++/doc/index.html) because I needed MGRS codes (I went down a rabbit hole and ended up in the old military code to a point, I only needed precision == 0 but got distracted, then found out MGRS is built in): 

https://github.com/hypertidy/geographiclib

as well as polygon area and perimeter and all other manner of stuff it has projections, here's a few of them: 

https://geographiclib.sourceforge.io/C++/doc/classGeographicLib_1_1AzimuthalEquidistant.html

https://geographiclib.sourceforge.io/C++/doc/classGeographicLib_1_1LambertConformalConic.html

https://geographiclib.sourceforge.io/C++/doc/classGeographicLib_1_1PolarStereographic.html

I found it really easy to wrap so will chip away as I need stuff. If you've got any pressing needs let me know.

**Comment by mdsumner** [comment]
it's super easy to do, I just added a basic copy/paste example from the api doc: 

https://github.com/hypertidy/geographiclib/commit/2d2e4edd2649ac13bd9f0e5fea7c8fccce51a770

**Comment by mdsumner** [comment]
there's also Rust geodesy which I've been following a little, I will explore this more too (this one has omerc, which geographiclib does not): 

https://github.com/busstoptaktik/geodesy/blob/main/src/inner_op/lcc.rs

**Comment by mpadge** [comment]
Cool, thanks for the update - geographiclib looks awesome.

**Comment by mdsumner** [comment]
geographiclib is in a lot better shape now see https://hypertidy.github.io/geographiclib/reference/polygon_area.html and https://hypertidy.github.io/geographiclib/reference/utmups_fwd.html  (I'm using MGRS codes extensively for batch lookup of Sentinel 2 sources without constantly running stac queries)
